src/server.c

```diff
@@ -6284,7 +6284,10 @@ void monitorCommand(client *c) {
     }
 
     /* ignore MONITOR if already slave or in monitor mode */
```

Comment:
Maybe this comment is not proper now

Reply:
I mean, technically we’re still ignoring monitor command when in monitor. I’m not familiar with what “slave” means in this context though

Reply:
For some reason, we re-use the logic for replication for monitor (although, at this point I think they are completing different mechanisms). So you would likely want to add a new check like:
```
if (c->flags & CLIENT_SLAVE) {
    if (c->flags & CLIENT_MONITOR) {
        addReplyError(whatever);
    } else {
        return;
    }
}
```